lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -7,6 +7,8 @@ class FileThreadPool
         DEFAULT_CONCURRENCY = 2
         MAX_CONCURRENCY = 2
 
+        Thread.abort_on_exception = true
```

Comment:
@wfleming: @maxjacobson and I confirmed that this change causes the OOM java error to get raised in the ruby thread. We used a toy example for testing and `Thread#join` didn't cause the exception to make it through.

errored build: https://codeclimate.com/repos/57e4d7aad0d6c36836004456/builds/9
beta channel branch: https://github.com/codeclimate/codeclimate-duplication/tree/channel/beta

Updating this branch